blog/executor-plugins.md

```diff
@@ -0,0 +1,348 @@
+
+```bash
+git add -A
+git commit -m "Implement gh executor"
+git tag v1.0.0
+```
+
+Next, let's push our changes and the new tag:
+
+```bash
+git push
+git push --tags
+```
+
+This triggers GitHub Action:
+
+![GitHub action that creates a new release](./assets/release-job.png)
+
+**What this automation does under the hood**
+
+This automation:
+
+1. Installs the latest [GoReleaser](https://goreleaser.com/) tool.
+2. Builds all plugin binaries defined in the `.goreleaser.yaml` file.
+3. Generates an index file using the [Botkube helper tool](https://docs.botkube.io/plugin/repo#generate-index-file).
+4. Generates a release description.
+5. Uses the [`gh`](https://cli.github.com) CLI to create a new GitHub release.
+
+### Use the `gh` executor
+
+In the description of a new GitHub release, you will see the repository URL that you can use within Botkube.
+
+#### Steps
+
+1. Follow one of our [installation guides](https://docs.botkube.io/installation). Once you reach the Botkube deployment step, add flags specified in the steps below to `helm install`.
+2. Export required environment variables:
+
+   :::info
+   Follow the official GitHub guide on how to create a [personal access token](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token#creating-a-personal-access-token-classic). To be able to create GitHub issues, add the `repo` permission.
+   :::
+
+   ```bash
+   export REPOSITORY={repo} # format OWNER/REPO_NAME, e.g. kubeshop/botkube
+   export GITHUB_TOKEN={token}
+   export PLUGINS_URL={plugin_index_url}
+   ```
+
+3. Add the `gh` executor related configuration:
+
+   ```bash
+   # example configuration for `gh` executor
+   -f https://gist.github.com/mszostok/88e6852f93429928d1183e52226921fc/raw \
+   --set 'plugins.repositories.botkube-plugins.url'=${PLUGINS_URL} \
+   --set 'executors.plugin-based.botkube-plugins/gh.config.github.repository'=${REPOSITORY} \
+   --set 'executors.plugin-based.botkube-plugins/gh.config.github.token'=${GITHUB_TOKEN} \
+   ```
+
+4. Depending on the selected platform, add the `plugin-based` executor binding. For Slack, it looks like this:
+
+   ```bash
+   --set communications.default-group.socketSlack.channels.default.bindings.executors=['plugin-based']
+   ```
+
+If you follow all the steps above, you will have all the necessary flags allowing you to install Botkube with the `gh` executor!
+
+Here's an example of a full command that you should have constructed for Slack installation:
+
+<Gist id="d8828c7eb80e72bba5fbbf442991eb7e" />
+
+### Testing
+
+1. Navigate to your communication channel
+2. On a given channel, run:
+
+   ```bash
+   @Botkube list executors
+   ```
+
+   It should return information about enabled `gh` executor:
+   ![](./assets/list-exec.png)
+
+3. Create a failing Job:
+
+   ```bash
+   kubectl create -f https://gist.github.com/mszostok/a07230db374e48b9f88491e197d708fd/raw
```

Comment:
I think it would be great to have it visible as you need to check what you're creating on the cluster anyway 🙂 
Something like this IMO would be better:

```
cat > /tmp/job.yaml << ENDOFFILE
#...
ENDOFFILE
```

I know, a few lines more, but still, it's for copy paste, so..